src/Catalog/Helpers/GalleryDatabaseQueryService.cs

```diff
@@ -167,7 +154,25 @@ public async Task<FeedPackageDetails> GetPackageOrNull(string id, string version
         internal static string BuildDb2CatalogSqlQuery(Db2CatalogCursor cursor)
         {
             return $@"SELECT TOP {cursor.Top} WITH TIES
-                        {Db2CatalogSqlSubQuery}
+                        PR.[Id],
+                        P.[NormalizedVersion],
+                        P.[Created],
+                        P.[LastEdited],
+                        P.[Published],
+                        P.[Listed],
+                        P.[HideLicenseReport],
+                        P.[LicenseNames],
+                        P.[LicenseReportUrl],
+                        PD.[Status] AS '{Db2CatalogProjectionColumnNames.DeprecationStatus}',
+                        APR.[Id] AS '{Db2CatalogProjectionColumnNames.AlternatePackageId}',
```

Comment:
When a package version is chosen instead of just the registration, `AlternatePackageRegistrationKey` will be null.
You need to fetch the package ID using `AlternatePackageKey`'s `PackageRegistrationKey`.

As a result, you need to pull in all the following things
`AlternatePackageRegistrationId`
`AlternatePackageId`
`AlternatePackageVersion`

Reply:
Now using the `IsNull` SQL construct on the join in https://github.com/NuGet/NuGet.Services.Metadata/pull/542/commits/8b9758d296cee6dc79a5f80535fab1f2aad67f7c

If no alternate package version is defined, `AP.[PackageRegistrationKey]` will yield `null`, and the join will happen on the `PackageDeprecation.AlternatePackageRegistrationKey` instead.